src/Components/test/testassets/BasicTestApp/Index.razor

```diff
@@ -66,8 +66,13 @@
         <option value="BasicTestApp.NavigationFailureComponent">Navigation failure</option>
         <option value="BasicTestApp.ParentChildComponent">Parent component with child</option>
         <option value="BasicTestApp.PropertiesChangedHandlerParent">Parent component that changes parameters on child</option>
-        <option value="BasicTestApp.ProtectedBrowserStorageUsageComponent">Protected browser storage usage</option>
-        <option value="BasicTestApp.ProtectedBrowserStorageInjectionComponent">Protected browser storage injection</option>
+        @if (!OperatingSystem.IsBrowser())
+        {
+            var serverAssembly = AppDomain.CurrentDomain.GetAssemblies().First(f => f.GetName().Name == "Components.TestServer");
+
+            <option value="Components.TestServer.ProtectedBrowserStorageUsageComponent, @serverAssembly.FullName">Protected browser storage usage</option>
+            <option value="Components.TestServer.ProtectedBrowserStorageInjectionComponent, @serverAssembly.FullName">Protected browser storage injection</option>
+        }
```

Comment:
I'm sure this is viable as-is, but if you feel like you have time to tweak it a bit more, I'd suggest it would be preferable not to have the `@if` statement, because then it would be really confusing if you were in the browser UI looking for something in the list that wasn't there because you accidentally loaded the wrong host environment and didn't realise some test cases are only available in some environments.

It would be much clearer if all the entries were present all the time, but if you picked a type that can't be found, we have an error message like "The type {name} could not be found. It might not be supported when running on this host environment.". This error could be triggered from `Index.razor`'s `SelectedComponentType` getter, or just inline in the Razor UI logic when `SelectedComponentTypeName` is nonempty but `SelectedComponentType` is null.

Also rather than using assembly qualified names, what stops it from working just having the type name (independently of the assembly)? The logic in `Index.razor`'s `SelectedComponentType` uses `Type.GetType` - doesn't that look at all the loaded assemblies?

Reply:
>  The logic in Index.razor's SelectedComponentType uses Type.GetType - doesn't that look at all the loaded assemblies?

It only works if the type is in the [executing assembly](https://docs.microsoft.com/en-us/dotnet/api/system.type.gettype?view=netcore-3.1#System_Type_GetType_System_String_) which requires the additional work around fully qualifying it. I have this tweaked a bit where we rely on the runtime to fail loudly if the type is unable to be found. Perhaps that would suffice here?

Reply:
Ah yes, makes sense.

The improvement you've added looks good to me. It's test code - we can always try to streamline it later if we want.